source/common/http/filter_manager.h

```diff
@@ -748,6 +750,9 @@ class FilterManager : public ScopeTrackedObject,
   void decodeHeaders(RequestHeaderMap& headers, bool end_stream) {
     state_.remote_decode_complete_ = end_stream;
     decodeHeaders(nullptr, headers, end_stream);
+    if (allow_upstream_half_close_) {
```

Comment:
can this be cone in maybeendDecode so we don't have to do it in all headers/body/data cases?  if maybeEndStream did checks on allow_upstream_half_close_ I think it'd simplify things?

Reply:
Moved the flag check to `checkAndCloseStreamIfFullyClosed`